app/bitcoin/bitcoin.js

```diff
@@ -28,35 +42,60 @@ class BitcoinAddress {
     _hasBitcoinAddress() {
         return this.persistence.hasBitcoinAddress();
     }
+}
 
-    checkBalance() {
-        const address = this.persistence.getPublicKey();
-        return this.serverRequests.getBalance(address);
+
+class TransactionInput {
+    constructor(tx, index, value) {
+        this.tx = tx;
+        this.index = index;
+        this.value = value;
+    }
+}
+
+
+class TransactionOutput {
+    constructor(recipient, amount) {
+        this.recipient = recipient;
+        this.amount = amount;
     }
 }
 
+
+//Instantiate BitcoinTransfer to begin building a list of inputs and outputs
 class BitcoinTransfer {
 
-    //TODO: Allow #buildTransaction() to take a list of sources
-    /**
-     * Sends an amount of bitcoin to the destination wallet
-     * @param sourcetx  - The transaction to use as a source for this transaction
-     * @param sourceIndex   - The index of the source output in its transaction
-     * @param to        - Public key of the recipient's bitcoin wallet
-     * @param from      - Private key of the sender's bitcoin wallet
-     * @param amount    - Amount of bitcoin to buildTransaction (in Satoshi)
-     */
-    static buildTransaction(sourcetx, sourceIndex, from, to, amount) {
-        const transaction = new bitcoin.TransactionBuilder();
+    constructor(myAddress) {
+        this.myAddress = myAddress;
+        this.inputs = [];
+        this.outputs = [];
+        this.currentInputValue = 0;
+        this.PERCENTAGE_CONSTANT = 0.2;
+        this.THRESHOLD = 0.000001;
+    }
+
+    addInput(input) {
```

Comment:
For `BitcoinTransfer`, it makes sense to make it a class, because it has these helper methods on it: `addInput`, `addOutput`, etc.